pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -225,10 +226,11 @@ private RelNode optimize(RelRoot relRoot, PlannerContext plannerContext) {
     }
   }
 
-  private QueryPlan toDispatchablePlan(RelRoot relRoot, PlannerContext plannerContext, long requestId) {
+  private QueryPlan toDispatchablePlan(RelRoot relRoot, PlannerContext plannerContext, long requestId,
+      Map<String, String> options) {
     // 5. construct a dispatchable query plan.
     StagePlanner queryStagePlanner = new StagePlanner(plannerContext, _workerManager, requestId, _tableCache);
-    return queryStagePlanner.makePlan(relRoot);
+    return queryStagePlanner.makePlan(relRoot, options);
```

Comment:
we can pass in the plannerContext as 3rd argument, or simply
```suggestion
    return queryStagePlanner.makePlan(relRoot, plannerContext.getOptions());
```